osu.Game/Screens/Select/BeatmapCarousel.cs

```diff
@@ -49,6 +49,11 @@ public partial class BeatmapCarousel : CompositeDrawable, IKeyBindingHandler<Glo
         /// </summary>
         public Action? BeatmapSetsChanged;
 
+        /// <summary>
+        /// Triggered when the deleted <see cref="BeatmapSets"/> change.
+        /// </summary>
+        public Action? DeletedBeatmapSetsChanged;
+
```

Comment:
Rather than adding a new event flow here, I'd recommend using `BeatmapSetsChanged` and adjusting the xmldoc accordingly (to something like "Triggered when beatmapsets finish loading, or are subsequently changed").

If you check the usages of `BeatmapSetsChanged`, there's basically one, and that path already has a well guarded update count call.